Content.Server/Magic/MagicSystem.cs

```diff
@@ -0,0 +1,269 @@
+    /// </summary>
+    /// <param name="args"></param>
+    private void OnTeleportSpell(TeleportSpellEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        var transform = Transform(args.Performer);
+
+        if (_mapManager.TryFindGridAt(args.Target, out var grid))
+        {
+            var gridPosition = grid.WorldToLocal(args.Target.Position);
+
+            transform.Coordinates = new EntityCoordinates(grid.GridEntityId, gridPosition);
+            SoundSystem.Play(Filter.Pvs(args.Target), args.BlinkSound.GetSound());
+        }
+        else
+        {
+            var mapEntity = _mapManager.GetMapEntityIdOrThrow(args.Target.MapId);
+            transform.WorldPosition = args.Target.Position;
+            transform.AttachParent(mapEntity);
+            SoundSystem.Play(Filter.Pvs(args.Target), args.BlinkSound.GetSound());
+        }
+
+        args.Handled = true;
+    }
+
+    /// <summary>
+    /// Spawns 3 walls in front of the caster in a 3x1/1x3 pattern
+    /// Disappears after a set amount of time
+    /// </summary>
+    /// <param name="args"></param>
+    private void OnForceWallSpell(ForceWallSpellEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        var transform = Transform(args.Performer);
+        var directionPos = transform.Coordinates.Offset(transform.LocalRotation.ToWorldVec().Normalized);
+
+        if (!_mapManager.TryGetGrid(transform.GridID, out var mapGrid))
+            return;
+        if (!directionPos.TryGetTileRef(out var tileReference, EntityManager, _mapManager))
+            return;
+
+        var tileIndex = tileReference.Value.GridIndices;
+        var coords = mapGrid.GridTileToLocal(tileIndex);
+        EntityCoordinates coordsPlus;
+        EntityCoordinates coordsMinus;
+
+        var dir = transform.LocalRotation.GetCardinalDir();
+        switch (dir)
+        {
+            case Direction.North:
+            case Direction.South:
+            {
+                coordsPlus = mapGrid.GridTileToLocal(tileIndex + (1, 0));
+                coordsMinus = mapGrid.GridTileToLocal(tileIndex + (-1, 0));
+                break;
+            }
+            case Direction.East:
+            case Direction.West:
+            {
+                coordsPlus = mapGrid.GridTileToLocal(tileIndex + (0, 1));
+                coordsMinus = mapGrid.GridTileToLocal(tileIndex + (0, -1));
+                break;
+            }
+            default:
+                return;
+        }
+
+        SoundSystem.Play(Filter.Pvs(coords), args.ForceWallSound.GetSound(), AudioParams.Default.WithVolume(args.ForceWallVolume));
+        ActiveWalls.Add(Spawn(args.WallPrototype, coords));
+        ActiveWalls.Add(Spawn(args.WallPrototype, coordsPlus));
+        ActiveWalls.Add(Spawn(args.WallPrototype, coordsMinus));
+
+        foreach (var wall in ActiveWalls)
+        {
+            EnsureComp<ForceWallSpellComponent>(wall);
+        }
```

Comment:
This should be on the prototype itself.

Reply:
Just to clarify. It should be on the Force Wall prototype in walls.yml? Just want to be sure because I was following how Invisible Walls worked for mime powers and it wasn't included in the yml there.

Reply:
I'm not familiar with mime code but it sounds sussy so I'll check it out later.

Reply:
Unless there's a bug in adding the component to the YAML, then it should be fine to just do that.